test/functional/tools/parameters/gx_allow_uri_if_protocol.xml

```diff
@@ -0,0 +1,13 @@
+<tool id="gx_allow_uri_if_protocol" name="gx_allow_uri_if_protocol" version="1.0.0">
+    <command><![CDATA[
+echo '$input' > '$output'
```

Comment:
Realistically tools will have to support both materialized and deferred datasets, since you can pass both. Can you spec out how tool authors should handle that? Alternatively you could pass a file:// uri, but in either case we should document what a tool needs to handle.

Reply:
Thanks for the good feedback!

I've expanded a bit the documentation in the XSD and the test tool. I hope this is what you had in mind, if not, let me know :+1: